marklogic-spark-connector/src/main/java/com/marklogic/spark/writer/DocumentRowConverter.java

```diff
@@ -82,17 +120,97 @@ private Iterator<DocBuilder.DocumentInputs> readContentFromRow(String uri, Inter
+        final int extractedTextRow = getExtractedTextRow(row);
+        if (extractedTextRow > 0) {
+            String extractedText = row.getString(extractedTextRow);
             DocBuilder.DocumentInputs extractedTextDocument = Format.XML.equals(this.extractedTextFormat) ?
                 buildExtractedXmlDocument(uri, extractedText, metadata) :
                 buildExtractedJsonDocument(uri, extractedText, metadata);
             return Stream.of(mainInputs, extractedTextDocument).iterator();
         }
+
+        final int classifiedTextRow = getClassifiedTextRow(row);
+        if (classifiedTextRow > 0) {
+            String classificationText = row.getString(classifiedTextRow);
+            String format = row.getString(2);
+            if ("XML".equals(format)) {
+                mainInputs = appendDocumentClassificationToXmlContent(new String(row.getBinary(1), StandardCharsets.UTF_8), uri, classificationText, uriTemplateValues, metadata);
+            } else if ("JSON".equals(format)) {
+                mainInputs = appendDocumentClassificationToJsonContent(uri, classificationText, uriTemplateValues, metadata);
+            }
+        }
+
         return Stream.of(mainInputs).iterator();
     }
 
+    private int getExtractedTextRow(InternalRow row) {
+        if (row.numFields() > 8) {
+            for (int i = 8; i < row.numFields(); i++) {
+                if (!columnIsClassifiedText(row.getString(i))) {
+                    return i;
+                }
+            }
+        }
+        return 0;
+    }
+
+    private int getClassifiedTextRow(InternalRow row) {
+        if (row.numFields() > 8) {
+            for (int i = 8; i < row.numFields(); i++) {
+                if (columnIsClassifiedText(row.getString(i))) {
+                    return i;
+                }
+            }
+        }
+        return 0;
+    }
+
+    private boolean columnIsClassifiedText(String text) {
+        return text.contains(CLASSIFICATION_MAIN_ELEMENT);
+    }
+
+    private DocBuilder.DocumentInputs appendDocumentClassificationToJsonContent(
+        String uri, String classificationText, JsonNode uriTemplateValues, DocumentMetadataHandle metadata
+    ) {
+        byte[] response = textClassifier.classifyTextToBytes(uri, classificationText);
+        try {
+            ObjectNode originalJson = (ObjectNode) objectMapper.readTree(classificationText);
+            JsonNode structuredDocumentNode = xmlMapper.readTree(response).get(CLASSIFICATION_MAIN_ELEMENT);
+            originalJson.set("classification", structuredDocumentNode);
+            return new DocBuilder.DocumentInputs(uri, new JacksonHandle(originalJson), uriTemplateValues, metadata);
+        } catch (IOException e) {
+            throw new ConnectorException(String.format("Unable to classify data from document with URI: %s; cause: %s", uri, e.getMessage()), e);
+        }
+    }
+
+    public DocBuilder.DocumentInputs appendDocumentClassificationToXmlContent(
```

Comment:
To keep things simple for now while we hash out the UDF approach, just support XML, ignore JSON for now. We'll need this logic for the following scenarios:

1. There's a "content" column and no "chunks" or "extractedText".
2. There's a "content" column and an "extractedText" column.
3. There's "content", "extractedText", and "chunks". 

We can do them one at a time, so this PR only needs to support the first one. But I'm thinking it'll be simpler to figure out how to support all 3 for just XML first, and then it'll be easy to add JSON support.

Reply:
> There's a "content" column and no "chunks" or "extractedText".
I'm not sure how to check for the existence of chunks.